### PR TITLE
Make hooks' "ignore_error" option work

### PR DESCRIPTION
**Pull request by julienduchesne**
It currently doesn't do anything

### CODE REVIEW

**Review by dblanchette — APPROVED** [review]

File: config/extension_hook.go [hunk 1]
```diff
@@ -180,7 +180,7 @@ func (list HookList) Run(status error, args ...interface{}) (result []interface{
 		currentErr = shell.FilterPlanError(currentErr, hook.options().TerraformCliArgs[0])
 		if _, ok := currentErr.(errors.PlanWithChanges); ok {
 			errs = append(errs, currentErr)
-		} else if currentErr != nil {
+		} else if currentErr != nil && !hook.IgnoreError {
```
Comment (jocgir) [inline config/extension_hook.go:183]: I think the original idea was to run the hook command even if an error occurred prior the error. From what I can see in the code, that is what is implemented.

I agree that the comment or name is wrong, but that could change the behavior.

**Review by AbdulRahmanAlHamali — APPROVED** [review]

File: config/extension_hook.go (hunk 1 above)
Comment (jocgir) [reply config/extension_hook.go:183]: The original goal was to run the cleanup at the end whether there is an error or not.

Comment (julienduchesne) [reply config/extension_hook.go:183]: Ah I see. The description isn't very clear `continue execution on error`

Comment (jocgir) [reply config/extension_hook.go:183]: The concept was first introduced on extra_command. But the meaning of this one was more execute even if there were error before...

Comment (jocgir) [reply config/extension_hook.go:183]: I think both concepts have values and this one should clearly renamed.

The error should not be ignored without leaving a trace in the logs though.

Comment (julienduchesne) [reply config/extension_hook.go:183]: It leaves a trace right now with this implementation

I think `ignore_error` should be what I'm trying to do and there could be a new option `run_on_errors` for the current behavior. What do you think?

Comment (jocgir) [reply config/extension_hook.go:183]: Yes precisely.

**Review by jocgir — APPROVED** [review]

**Pull request merged**